runtime/cmd/execute/debugger.go

```diff
@@ -0,0 +1,185 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package execute
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/c-bata/go-prompt"
+
+	"github.com/onflow/cadence/runtime/interpreter"
+)
+
+const commandShortHelp = "h"
+const commandLongHelp = "help"
+const commandShortContinue = "c"
+const commandLongContinue = "continue"
+const commandShortNext = "n"
+const commandLongNext = "next"
+const commandLongExit = "exit"
+const commandShortShow = "s"
+const commandLongShow = "show"
+const commandShortWhere = "w"
+const commandLongWhere = "where"
+
+var debuggerCommandSuggestions = []prompt.Suggest{
+	{Text: commandLongContinue, Description: "Continue"},
+	{Text: commandLongNext, Description: "Next / step"},
+	{Text: commandLongWhere, Description: "Location info"},
+	{Text: commandLongShow, Description: "Show variable(s)"},
+	{Text: commandLongExit, Description: "Exit"},
+	{Text: commandLongHelp, Description: "Help"},
+}
+
+type InteractiveDebugger struct {
+	debugger *interpreter.Debugger
+	stop     interpreter.Stop
+}
+
+func NewInteractiveDebugger(debugger *interpreter.Debugger, stop interpreter.Stop) *InteractiveDebugger {
+	return &InteractiveDebugger{
+		debugger: debugger,
+		stop:     stop,
+	}
+}
+
+func (d *InteractiveDebugger) Continue() {
+	d.debugger.Continue()
+}
+
+func (d *InteractiveDebugger) Next() {
+	d.stop = d.debugger.Next()
+}
+
+// Show shows the values for the variables with the given names.
+// If no names are given, lists all non-base variables
+//
+func (d *InteractiveDebugger) Show(names []string) {
+	current := d.debugger.CurrentActivation(d.stop.Interpreter)
+	switch len(names) {
+	case 0:
+		for name := range current.FunctionValues() { //nolint:maprangecheck
+			println(name)
```

Comment:
Any special reason to use built-in `println()` here and in other places, as opposed to `fmt.Println()`?

Reply:
No particular reason, I hadn't thought of `fmt.Println`. I refactored it everywhere in fd2a2d4de3d3f0cb73294b87d49b2336ae196130